src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerMetrics.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB
 {
-    internal class CosmosDBTriggerMetrics : ScaleMetrics
+    public class CosmosDBTriggerMetrics : ScaleMetrics
```

Comment:
This should be internal, not public

Reply:
And this needs to be public? Who will be consuming it? Won't normal users see the type and have access to it?

Reply:
Yes, this needs to be public for CosmosDBScaleMonitor to be public -- see below:
![image](https://user-images.githubusercontent.com/33586225/200038598-6400c73b-12af-4cf6-83b8-97dad1f79d73.png)

Users would be able to see the type in this case

Reply:
In the scale controller redesign, we call the GetMetricsAsync and GetScaleStatus methods directly, see [here](https://msazure.visualstudio.com/One/_git/AAPT-Antares-ScaleController?path=/src/ScaleController/Services/SiteScaleMonitorService.cs&version=GBv3.x&line=79&lineEnd=79&lineStartColumn=16&lineEndColumn=81&lineStyle=plain&_a=contents)

Reply:
But you need the interface to be public (ScaleMetrics) not the implementation.

Reply:
Copying notes from offline conversation with Matias: 

Ahh got it, I see the concern. From the scale controller redesign perspective, we do not need access to specific attributes in the CosmosDBTriggerMetrics class, we just need to be able to call GetScaleStatus and GetMetricsAsync. If we are able to rely on the generic implementations of these methods (for example [here](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/806/files#diff-87779aaf26d7ac619b8e357dd921c23dbd1ab7b725ac337a30b6fb26e8072046:~:text=%7D-,async%20Task%3CScaleMetrics%3E%20IScaleMonitor.GetMetricsAsync(),-%7B)), then we do not have to make the metrics class public. This would also entail changing the implemented class for ScaleMonitor from IScaleMonitor<CosmosDBTriggerMetrics> to IScaleMonitor<ScaleMetrics> https://github.com/Azure/azure-webjobs-sdk-extensions/pull/806/files#diff-87779aaf26d7ac619b8e357dd921c23dbd1ab7b725ac337a30b6fb26e8072046:~:text=class%20CosmosDBScaleMonitor%20%3A-,IScaleMonitor%3CCosmosDBTriggerMetrics%3E,-%7B

Reply:
Why were these made public again? I thought the interfaces/base types provided enough access, these do not need to be public

Reply:
You just need to work with `ScaleMetrics` or `ITargetScaler` or `IScaleMonitor`, you don't need `CosmosDBMetrics`.

Reply:
Created a scaler factory class which returns a generic `IScaleMonitor` implementation